arduino-core/src/processing/app/helpers/CommandlineParser.java

```diff
@@ -84,6 +85,10 @@ public void parseArgumentsPhase1() {
           }
           libraryToInstall = args[i];
         }
+        if (a == ACTION.VERSION) {
+          BaseNoGui.showMessage("Arduino", BaseNoGui.VERSION_NAME_LONG);
+          System.exit(0);
```

Comment:
I'm not sure I like this exit here, this bypasses any cleanup that would normally (perhaps) run after this function returns. Also, this function is expected to parse arguments only, not act on them. Also, passing `--version --verify` now just acts on `--version`, instead of showing an error that only one of them can be passed.

Better would seem to set `action` here, and act on it elsewhere.

Reply:
You're right!
Thank you for your comment.